### PR TITLE
Update azure node related methods in platform_nodes and azure_utils

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
This PR includes: 
 - Add support for force VM stop
 - Handle the vms/nodes list in low-level modules similar to what we do for other platforms

Also, Fixes: #3736 

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

**Comment by sidhant-agrawal** [comment]
This PR verification can be done once PR #3637 is merged.

**Review by suchita-g — CHANGES_REQUESTED** [review]

File: ocs_ci/utility/azure_utils.py
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py:359]: ack, will fix this.

File: ocs_ci/ocs/platform_nodes.py
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1724]: @ebenahar In other platforms, we are using `force=True` by default.
Should we keep `force=True` by default in Azure or use default value False according to API doc?

File: ocs_ci/utility/azure_utils.py
Comment (ebenahar) [inline ocs_ci/utility/azure_utils.py:333]: as discussed, let's set it here to False, to be aligned with Azure API default

Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated

File: ocs_ci/utility/azure_utils.py
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated

File: ocs_ci/ocs/platform_nodes.py
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1724]: as discussed offline, keeping `force=True` here

File: ocs_ci/ocs/platform_nodes.py
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1732]: as discussed offline, keeping `force=True` in platform_nodes.py

File: ocs_ci/utility/azure_utils.py
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated azure_utils.py

File: ocs_ci/ocs/platform_nodes.py
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1808]: as discussed offline, keeping `force=True` here

**Review by ebenahar — DISMISSED** [review]

**Review by suchita-g — DISMISSED** [review]
LGTM

**Comment by sidhant-agrawal** [comment]
Do not merge because it needs a small change after PR #3637 is merged.

**Review by suchita-g — APPROVED** [review]
LGTM

File: ocs_ci/utility/azure_utils.py
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py:312]: @mbukatov 
`restart_vm_instances()` will be called from `restart_nodes()` in platform_nodes.py.
Nodes status checks will be taken care of in that function.

**Review by ebenahar — APPROVED** [review]

**Pull request merged**
